### DISCUSSION TITLE
How about yarn? And why npm@7?

### DISCUSSION THREAD

**Discussion by liyachun01** [amplication/amplication#1176] [General]
Hello, I am a NestJS user in China, currently on an [Web + Electron App](https://www.gaoxiazhitu.com/). Interested at your work, and just starting the reading of your code, and may be ready for some contributions.

This may not be feature oriented but -- How about switching to Yarn or Yarn v2 for some smoother workspace management? In case you are not familiar, here are some good parts which We've been enjoying all along:

1. Yarn has it's own monorepo support called [Workspaces](https://yarnpkg.com/features/workspaces), and is able to [integrate](https://yarnpkg.com/features/workspaces#yarn-workspaces-vs-lerna) with lerna (read more [here](https://github.com/lerna/lerna/pull/899), or, I happen to have a post on Yarn v1 and lerna [here](https://dev.to/liyachun/javascript-monorepo-by-lerna-with-yarn-workspaces-and-git-submodules-4189)). For brevity:

  [root]/package.json
  ``` json
  {
    "private": true,
    "workspaces": {
        "packages": ["workspace-a", "workspace-b", "services/*"],
        "nohoist": ["**/react-native", "**/react-native/**"]
      }
  }
  ```
  [root]/lerna.json
  ``` json
  {
    "npmClient": "yarn",
    "useWorkspaces": true
  }
  ```
  Now you can simply omit Lerna's 'packages' setting and use Yarn's.

2. The Workspaces feature also make `npm install` life saving: a `yarn` in the root directory, will simply act like a combination of `npm install` + `lerna bootstrap`, but it will hoist cross package deps to the root node_modules, saving disk space (you have a nohoist option if you need).
3. [Selective Version Resolution](https://classic.yarnpkg.com/en/docs/selective-version-resolutions/): incase you have some across package deps and want to keep a consistent actual installed version, or encounter version conflicts, add this line helps: 

  [root]/package.json
  ``` json
  {
      "resolutions": {
        "@types/node": "^14.14.17",
        "typescript": "^4.1.3",
      }
  }
  ```
4. The above features actually already exist since v1.x somehow, but v2 is an epic change with a new installation strategy called [Plug'n'Play](https://yarnpkg.com/features/pnp), you still can fallback to the node_modules strategy for compatibility concerns meanwhile enjoy the improvments in performance.

Or if you already have your opinions? I noticed that you are using npm@7 which is not the LTS version. Willing to hear about it.

**Comment by yuval-hazaz** [comment]
@liyachun01 thanks for all the details.
We are using npm solely based on previous experience.
I am looking at yarn from time to time and wanted to consider the move but didn’t get the resources until now.
I would be happy to discuss the details to understand the impact before we kick of such a move
